### PR TITLE
CC-1257: Use `course-sdk lint` in `test-course-definition.yml`

### PR DESCRIPTION
**Pull request by ryan-gang**

### CODE REVIEW

**Comment by ryan-gang** [comment]
Keeping the old lint steps, and the new `course-sdk lint` step side by side for some time. Will remove it later on.

**Comment by ryan-gang** [comment]
Job run here: https://github.com/codecrafters-io/build-your-own-redis/actions/runs/9361405181/job/25771106648

File: .github/workflows/test-course-definition.yml [hunk 1]
```diff
@@ -55,6 +55,33 @@ jobs:
...
+
+      - uses: actions/checkout@v3
+        with:
+          path: courses/${{github.event.repository.name}}
+
+      - run: bun install
+      - run: make install
+
+      - name: "Lint using course-sdk"
+        run: course-sdk lint
```
Comment (rohitpaulk) [inline .github/workflows/test-course-definition.yml:82]: Shouldn't we also remove the other jobs in this file that are now replaced by `course-sdk` lint?

**Review by rohitpaulk — APPROVED** [review]

File: .github/workflows/test-course-definition.yml (hunk 1 above)
Comment (ryan-gang) [reply .github/workflows/test-course-definition.yml:82]: https://github.com/codecrafters-io/course-sdk/pull/15#issuecomment-2146739618

**Pull request merged**